File: Mandelbrot Projects/Mandelbrot Controlls/Main/Complex.java
public class Complex {
    float re;
    float im;
    
    Complex(float re, float im) {
        this.re = re;
        this.im = im;
    }

    static Complex mult(Complex c1, Complex c2) {
        float cRe = c1.re * c2.re - c1.im * c2.im;
        float cIm = c1.re * c2.im + c1.im * c2.re;
        
        Complex c = new Complex(cRe, cIm);

        return c;
    }

    void mult(Complex c2) {
        float cRe = re * c2.re - im * c2.im;
        float cIm = re * c2.im + im * c2.re;

        this.re = cRe;
        this.im = cIm;
    }

    static Complex add(Complex c1, Complex c2) {
        float cRe = c1.re + c2.re;
        float cIm = c1.im + c2.im;

        Complex c = new Complex(cRe, cIm);

        return c;
    }

    void add(Complex c2) {
        float cRe = re + c2.re;
        float cIm = im + c2.im;

        this.re = cRe;
        this.im = cIm;
    }

    float mag() {
        return (float) Math.sqrt(re * re + im * im);
    }

    @Override
    public String toString() {
        return re + " + " + im + "i - " + mag();
    }
}

File: Newton's Fractal/p5js/complex.js
class Complex {
    constructor(re, im) {
        this.re = re;
        this.im = im;
    }

    add(c) {
        return new Complex(this.re + c.re, this.im + c.im);
    }

    sub(c) {
        return new Complex(this.re - c.re, this.im - c.im);
    }

    mult(c) {
        return new Complex(this.re * c.re - this.im * c.im, this.re * c.im + this.im * c.re);
    }

    div(c) {
        let r1 = cabs(this)
        let r2 = cabs(c)
        let r = r1 / r2

        let theta1 = Math.atan2(this.im, this.re)
        let theta2 = Math.atan2(c.im, c.re)

        return new Complex(r * Math.cos(theta1 - theta2), r * Math.sin(theta1 - theta2))
    }

    pow(n) {
        let r = cabs(this) ** n;
        let theta = n * Math.atan2(this.im, this.re);
        return new Complex(r * Math.cos(theta), r * Math.sin(theta));
    }
}

function cabs(c) {
    return Math.sqrt(c.re * c.re + c.im * c.im);
}

function cabs2(c) {
    return c.re * c.re + c.im * c.im;
}